src/main/java/org/learntotravel/web_ui/repository/DirectionRepositoryListImpl.java

```diff
@@ -0,0 +1,64 @@
+package org.learntotravel.web_ui.repository;
+
+import org.learntotravel.web_ui.direction.Direction;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Component("DirectionRepositoryList")
+public class DirectionRepositoryListImpl implements DirectionReposiotry {
+
+    private List<Direction> directions;
+    private volatile long latest;
```

Comment:
@Dzianiskulesh Are you sure about volatile keyword? Probably it's worth it to make method addDirection synchronized?